proto/src/proto.rs

```diff
@@ -2544,38 +2572,47 @@ impl From<LdapFilter> for Tag {
             LdapFilter::Extensible(f) => Tag::Sequence(Sequence {
                 id: 9,
                 class: TagClass::Context,
-                inner: f
-                    .matching_rule
-                    .into_iter()
-                    .map(|v| {
-                        Tag::OctetString(OctetString {
+                inner: {
+                    let mut res = vec![];
+                    let LdapMatchingRuleAssertion {
+                        matching_rule,
+                        type_,
+                        match_value,
+                        dn_attributes,
+                    } = f;
+
+                    match matching_rule {
+                        Some(v) => res.push(Tag::OctetString(OctetString {
                             inner: Vec::from(v),
                             id: 1,
                             class: TagClass::Context,
-                        })
-                    })
-                    .chain(f.type_.into_iter().map(|v| {
-                        Tag::OctetString(OctetString {
+                        })),
+                        None => {}
+                    }
```

Comment:
Suggestion: you can use `if let Some(v) = matching_rule` if you just need to handle one specific pattern.

Reply:
updated